shell/platform/fuchsia/flutter/text_delegate.h

```diff
@@ -0,0 +1,152 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Text editing functionality delegated from |PlatformView|.
+/// See |TextDelegate| for details.
+
+#ifndef FLUTTER_SHELL_PLATFORM_FUCHSIA_TEXT_DELEGATE_H_
+#define FLUTTER_SHELL_PLATFORM_FUCHSIA_TEXT_DELEGATE_H_
+
+#include <memory>
+
+#include <fuchsia/ui/input/cpp/fidl.h>
+#include <fuchsia/ui/input3/cpp/fidl.h>
+#include <fuchsia/ui/views/cpp/fidl.h>
+#include <lib/fidl/cpp/binding.h>
+
+#include "flutter/lib/ui/window/platform_message.h"
+#include "flutter/shell/common/platform_view.h"
+#include "flutter/shell/platform/fuchsia/flutter/keyboard.h"
+
+#include "logging.h"
+
+namespace flutter_runner {
+
+/// The channel name used for text editing platofrm messages.
+constexpr char kTextInputChannel[] = "flutter/textinput";
+
+/// The channel name used for key event platform messages.
+constexpr char kKeyEventChannel[] = "flutter/keyevent";
+
+/// TextDelegate handles keyboard inpout and text editing.
+///
+/// It mediates between Fuchsia's input and Flutter's platform messages. When it
+/// is initialized, it contacts `fuchsia.ui.input.Keyboard` to register itself
+/// as listener of key events.
+///
+/// Whenever a text editing request comes from the
+/// Flutter app, it will activate Fuchsia's input method editor, and will send
+/// text edit actions coming from the Fuchsia platform over to the Flutter app,
+/// by converting FIDL messages (`fuchsia.ui.input.InputMethodEditorClient`
+/// calls) to appropriate text editing Flutter platform messages.
+///
+/// For details refer to:
+///   * Flutter side:
+///   https://api.flutter.dev/javadoc/io/flutter/embedding/engine/systemchannels/TextInputChannel.html
+///   * Fuchsia side: https://fuchsia.dev/reference/fidl/fuchsia.ui.input
+class TextDelegate : private fuchsia::ui::input3::KeyboardListener,
```

Comment:
public inheritance pls
https://google.github.io/styleguide/cppguide.html#Inheritance

Reply:
The private inheritance predates this refactor, but OK.